pkg/diagnose/connectivity/trace_request.go

```diff
@@ -24,10 +24,74 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/config/resolver"
 	"github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
 )
 
+func init() {
+	diagnosis.Register("connectivity-datadog-core-endpoints", diagnose)
+}
+
+func diagnose(diagCfg diagnosis.DiagnoseConfig) []diagnosis.Diagnosis {
+
+	// Create domain resolvers
+	keysPerDomain, err := utils.GetMultipleEndpoints(config.Datadog)
+	if err != nil {
+		return []diagnosis.Diagnosis{
+			{
+				Result:      diagnosis.DiagnosisSuccess,
+				Name:        "Endpoints configuration",
+				Diagnosis:   "Misconfiguration of agent endpoints",
+				Remediation: "Please validate Agent configuration",
+				RawError:    err,
+			},
+		}
+	}
+
+	diagnoses := make([]diagnosis.Diagnosis, 0)
+
+	domainResolvers := resolver.NewSingleDomainResolvers(keysPerDomain)
+
+	client := forwarder.NewHTTPClient(config.Datadog)
+
+	// Send requests to all endpoints for all domains
+	for _, domainResolver := range domainResolvers {
+		// Go through all API Keys of a domain and send an HTTP request on each endpoint
+		for _, apiKey := range domainResolver.GetAPIKeys() {
+
+			for _, endpointInfo := range endpointsInfo {
+
+				domain, _ := domainResolver.Resolve(endpointInfo.Endpoint)
+
+				ctx := context.Background()
```

Comment:
does every request to an endpoint need a new context? Or can we just define a context at the beginning of the function and re-use it throughout?

Reply:
I fee that new context needed at least on the face of it since it is formed from the inner loop object. Perhaps better informed Agent developer like @sgnn7 or @truthbk  can suggest if this is possible.

Reply:
Given that it doesn't seem like there'll be concurrency across requests I think using a single `ctx` might be OK, actually. We don't do much with the context even in the face of a timeout or whatever either.

Reply:
I have looked at the wrong variable endpointInfo for some reason. If course ctx should be moved out. Done (in the next commit).

Reply:
LGTM 👍